python/python/bystro/vcf_utils/sim_data_generator.py

```diff
@@ -0,0 +1,61 @@
+"""VCF index and data simulator for testing analysis modules."""
+
+import random
+from io import StringIO
+
+import pandas as pd
+
+HEADER_COLS = ["#CHROM", "POS", "ID", "REF", "ALT", "QUAL", "FILTER", "INFO", "FORMAT"]
+
+
+def generate_random_vcf_index():
+    """Generate autosomal chr vcf variant IDs to use for testing."""
+    random_chr = random.randint(1, 22 + 1)
+    random_pos = random.randint(1, 1000000)
+    random_ref = random.choice(["A", "T", "C", "G"])
+    random_alt = random.choice([letter for letter in ["A", "T", "C", "G"] if letter != random_ref])
+    return f"{random_chr}:{random_pos}:{random_ref}:{random_alt}"
+
+
+def generate_simulated_vcf(
+    num_samples:int, num_vars:int
+) -> tuple[pd.DataFrame, str]:
+    """VCF simulator for testing analysis modules."""
+    # This is a first pass - could include more sophisticated choices for possible values in future
+    sample_ids = [f"SampleID{i+1}" for i in range(num_samples)]
+    header = HEADER_COLS + sample_ids
+    vcf_data = []
+    simulated_indices = []
+    vcf_data.append("\t".join(header))
+    for position in range(num_vars):
+        chrom = "chr1"
+        pos = position
+        variant_id = generate_random_vcf_index()  # Using simulated index as variant ID
+        simulated_indices.append(variant_id)  # Store the simulated index
+        ref = random.choice(["A", "T", "C", "G"])
```

Comment:
I think I'm lost here-- why are we regenerating info that we've already created with `generated_random_vcf_index`?